extension/src/Package.ts

```diff
@@ -286,6 +287,20 @@ function uriJoin(directory: vscode.Uri, file: string) {
 
 function findVsixFile(manifest: PackageManifest) {
     if (manifest.files) {
+        if (manifest.osSpecific) {
+            const os = require("os");
+            const osEntry = manifest.osSpecific[os.platform()];
+
+            if (osEntry) {
+                const osSpecificFile = manifest.files.find(e => (e === osEntry && typeof e === 'string' && e.endsWith('.vsix')));
```

Comment:
Is this whole check necessary? We can probably just assume that if you've listed an OS-specific VSIX file, that's the file you want. I can't see any particular reason we need to verify that it is also in `files` and has a .vsix extension.

Reply:
You are right, it's unnecessary double checking.